lib/wordpress.js

```diff
@@ -28,7 +28,7 @@ function parseArguments( args ) {
 
 function Client( settings ) {
 	[ "url", "username", "password" ].forEach(function( prop ) {
-		if ( !settings.hasOwnProperty( prop ) ) {
+		if ( !settings.hasOwnProperty( prop ) || settings[prop] === "" ) {
```

Comment:
If you're going to check for empty values, than you should handle `null` and `undefined` as well.  Just reduce the check to `if ( !settings[ prop ] )`.